content/guide/core/file-system.md

```diff
@@ -0,0 +1,747 @@
+ 
+  })
+  .catch(error => {
+    console.log(error)
+  })
+
+```
+- To read data from a text file, use the `readText` or `readTextSync` method of the file instance:
+```ts
+   file.readText().then(res => {
+      // Succeeded read from file.
+      
+    })
+    .catch(error => {
+
+  })
+```
+### Check if a file exists
+
+To check if a file exists, call the static [exists](#exists) method of the File class and pass it the file path string:
+
+```ts
+const exists = File.exists(fPath)
+```
+
+### Renaming a file
+
+To rename a file, use the `rename` or `renameSync` method: 
+
+```ts
+file.rename("new-name.ext").then((value)=>{
+
+}).catch(err=>{
+
+})
+```
+
+### Saving a binary data to a file
+<!-- TODO: binary data example -->
+To save binary data to a file, use the `write` or `writeSync` method:
+
+```ts
+file
+  .write(binary data)
+  .then(result => {
+    // Succeeded writing to the file.
+ 
+  })
+  .catch(error => {
+    console.log(error)
+  })
+
+```
+
+
+### Reading a binary data from a file
+
+To read binary data, use the `read` or `readSync` method of the File instance. For example, you can save an image on the device and retrive it as follows:
+
+```ts
+async readAnImage() {
+
+    const folder = knownFolders.documents()
+    const fPath = path.join(folder.path, 'cat.png')
+    const imageFile = File.fromPath(fPath)
+
+    try {
+
+      const imageSource = await ImageSource.fromFile('~/assets/images/download.jpeg')
+      const saved: boolean = imageSource.saveToFile(fPath, "png")
+
+      if (saved) {
+        // READ BINARY DATA
+        const imageBinary = await imageFile.read()
+        const img = await ImageSource.fromData(imageBinary)
+      }
+
+    } catch (err) {
+
+    }
+}
+```
+
+### Deleting a file
+
+To remove a file, use the [remove](#remove) or [removeSync](#removesync) method of the File instance:
+
+```ts
+file
+  .remove()
+  .then((res: boolean) => {
+    // Success removing the file.
+  })
+  .catch(err => {
+  })
+ ``` 
+
+ ### Normalizing a path
+
+ To normalize a path use the `normalize` or create it with the `join` method from the [path](#path-operations) object:
+
+```ts
+const testPath = '///test.txt'
+let documentsFolder = knownFolders.documents()
+path.normalize(documentsFolder.path + testPath)
+```
+
+## FileSystem API
+
+### documents()
+
+```ts
+const folder: Folder = knownFolders.documents()
+```
+
+Gets the Documents folder available for the current application. This Folder is private for the application and not accessible from Users/External apps.
+
+---
+
+### externalDocuments()
+
+```ts
+const folder: Folder = knownFolders.externalDocuments()
+```
+
+Gets the Documents folder available for the current application on an external storage. This Folder is private for the application and not accessible from Users/External apps. On android this requires `READ_EXTERNAL_STORAGE/WRITE_EXTERNAL_STORAGE` permissions. There is no external storage on iOS os it is the same as [documents()](#documents).
```

Comment:
Gets the Documents folder available for the current application from an external storage source. This folder has private access, with availability limited to the application, meaning it is not accessible to outside users, bots or external apps. 

On Android, for such read or write access, the flags `READ_EXTERNAL_STORAGE/WRITE_EXTERNAL_STORAGE` permissions need to be set to true.  This can be done by adding the following XML code to the AndroidManifest.xml file ->
  <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
  <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />

. There is no external storage on iOS, it is the same as [documents()](#documents).